File: 1310/Lucro.cpp
using namespace std;
#include <iostream>
#include <vector>
#include<bits/stdc++.h>
#include <algorithm>


int kadane(vector<int>& list){
    
    int max_cust = list[0];
    int max_current = list[0];
    
    for (int i = 1; i < list.size(); i++){
        //cout << max_current << " " << max_cust << endl;
        max_current = max(max_current+list[i],list[i]);
        max_cust = max(max_current, max_cust);
    }
    
    
    return max_cust;
}
int main(){
    int n, custo, k;
    vector<int> numbers;
    while (cin>>n){
        cin >> custo;
        //scout << custo;
        for (int i = 0; i < n; i++){
            cin >> k;
            numbers.push_back(k-custo);
        }
        
        
        int value = kadane(numbers);
        
        if (value <=0){
            cout << 0 << endl;
        } else{
            cout << value << endl;
        }
        
        numbers.clear();
    }
}

File: 1310/Lucro.py
def kadane(numbers):
    max_cust = current_cust = numbers[0]
    

    for i in numbers[1:]:
        current_cust = max(current_cust+i,i)
        max_cust = max(max_cust, current_cust)

    return max_cust

while True:

    try:
        n = int(input())

        custo = int(input())
        val = []
        for x in range(n):
            val.append(int(input())-custo)


        value = kadane(val)
        if value <= 0:
            print("0")
        else:
            print(value)
        





    except EOFError:
        break
